modules/standard/CTypes.chpl

```diff
@@ -1096,7 +1100,8 @@ module CTypes {
     :arg x: the by-reference argument to get a pointer to. Domains are not
             supported, and will cause a compiler error. Records, class
             instances, integral, real, imag, and complex types are supported.
-            For arrays, strings, or bytes, separate overloads should be used.
+            For arrays, strings, bytes, or class variables, separate overloads
+            should be used.
```

Comment:
I realize this is pre-existing, but since I'm looking at it for the first time:  What is the implication of the "separate overloads should be used" for the user?  They don't really get to select between the overloads, do they—the compiler just does that for them?

Reply:
As far as I know there's no implication for the user, and they can't select between overloads (it's just based on what type they use).

I guess really it would make more sense to only document one `c_ptrTo` overload (probably the generic type one), and in that doc describe all its special behavior cases; what would you think about that?

Reply:
Yeah, so if we kept the current code structure (which I can absolutely get behind), I'd probably either (a) expose all the overloads we think a user needs to know about and explain them all in a single comment before the first one with the user understanding that whatever type they pass in, one of the overloads following that description is going to take care of it and give them the behavior described here; or (b) find a way to rephrase this such that it doesn't suggest it's the user's responsibility to select an overload in some way.

But if we were to reorganize/refactor the code in some way, then my thinking had been similar to yours:  That maybe we just have one completely generic overload that we document and show the user; and then maybe the current type-oriented overloads become internal/private routines that we use to implement it (or we change the routine itself into a  big param/type conditional that sorts the cases into the appropriate branches).

I can get behind either approach and am not trying to make work for you here.  The phrase "separate overloads should be used" just suggested to me (pretending to be a user) that somehow I would do something to opt into specific overloads, which seems misleading and potentially confusing.

Reply:
As far as I understand we wouldn't have to do any code refactoring to make the doc change I suggested. We can just expose only our existing `c_ptrTo(ref x:?t)` and document what `c_ptrTo` does in all different type cases. The fact we actually implement the different behavior with distinct overloads can be considered an implementation detail.

Reply:
That said, that would require some thoughtful consideration into phrasing and I might not want to do that at the moment. I'd rather go with option b) for now, and perhaps just say "See overloads for arrays, strings, bytes, class variables which provide special behavior for those types".

Reply:
> As far as I understand we wouldn't have to do any code refactoring to make the doc change I suggested.

Oh I see... I didn't have the code in front of me and didn't understand/remember that there was a single most-generic overload.  Yeah, I think you're right that it could be done that way.  I'm fine with approach (b) though as well.

Reply:
I've gone ahead with (b) for expediency, but created https://github.com/Cray/chapel-private/issues/5386 to record this future desire as I'd really like to see it happen post-2.0.